Testing/Functional/Products/TestPlans/powerplatform.testplan.yaml

```diff
@@ -0,0 +1,116 @@
+ProductName: powerplatform
+TestPlan:
+  - PolicyId: MS.POWERPLATFORM.1.1v1
+    TestDriver: RunScuba
+    Tests:
+      - TestDescription: MS.POWERPLATFORM.1.1v1 Non-Compliant case - Allows production/sandbox environment creation by nonadmin users
+        Preconditions:
+          - Command: Set-TenantSettings
+            Splat:
+              RequestBody:
+                DisableEnvironmentCreationByNonAdminUsers: false
+        Postconditions: []
+        ExpectedResult: false
+      - TestDescription: MS.POWERPLATFORM.1.1v1 Compliant case - Restricts production/sandbox environment creation to admin users
+        Preconditions:
+          - Command: Set-TenantSettings
+            Splat:
+              RequestBody:
+                DisableEnvironmentCreationByNonAdminUsers: true
+        Postconditions: []
+        ExpectedResult: true
+
+  - PolicyId: MS.POWERPLATFORM.1.2v1
+    TestDriver: RunScuba
+    Tests:
+      - TestDescription: MS.POWERPLATFORM.1.2v1 Non-Compliant case - Allows trial environment creation by nonadmin users
+        Preconditions:
+          - Command: Set-TenantSettings
+            Splat:
+              RequestBody:
+                DisableTrialEnvironmentCreationByNonAdminUsers: false
+        Postconditions: []
+        ExpectedResult: false
+      - TestDescription: MS.POWERPLATFORM.1.2v1 Compliant case - Restricts trial environment creation to admin users
+        Preconditions:
+          - Command: Set-TenantSettings
+            Splat:
+              RequestBody:
+                DisableTrialEnvironmentCreationByNonAdminUsers: true
+        Postconditions: []
+        ExpectedResult: true
+
+  - PolicyId: MS.POWERPLATFORM.2.1v1
+    TestDriver: RunScuba
+    Tests:
+      - TestDescription: MS.POWERPLATFORM.2.1v1 Non-compliant - no DLP policy in default environment
+        Preconditions:
+          - Command: '(Get-DlpPolicy).value | Where {$_.environments.name -eq (Get-AdminPowerAppEnvironment -Default).EnvironmentName} | Select-Object @{Name="PolicyName"; Expression={$_.name}} | Remove-DlpPolicy'
+        Postconditions: []
+        ExpectedResult: false
+      - TestDescription: MS.POWERPLATFORM.2.1v1 Compliant - create DLP policy in default environment
+        Preconditions:
+          - Command: 'New-AdminDlpPolicy -DisplayName "DLP functional test" -EnvironmentName ((Get-AdminPowerAppEnvironment -Default).EnvironmentName)'
+        Postconditions: []
+        ExpectedResult: true
+
+  - PolicyId: MS.POWERPLATFORM.2.2v1
+    TestDriver: RunScuba
+    Tests:
+      - TestDescription: MS.POWERPLATFORM.2.2v1 Not Checked
+        Preconditions: []
+        Postconditions: []
+        IsNotChecked: true
+        ExpectedResult: true
```

Comment:
It was actually set to false, and was changed to true because Isabella found out that When the expected result is set to false it will still trigger the error message for some reason. Think David point out the issue. Any suggestion on how to fix it? 

![MicrosoftTeams-image](https://github.com/cisagov/ScubaGear/assets/107067015/d7922823-1f83-47bb-9414-8d50a16fabe7)

Reply:
The IsNotChecked flag is only for policies not checked by Scuba (e.g., MS.POWERPLATFORM.3.2v1  "Criticality" : "Should/Not-Implemented").  MS.POWERSHELL.2.2v1 is actally implemented and will require an implemented test case.

Reply:
My recommendatin would be to use 'RunCached' as your test driver for now.  When the database issue is resolved we can revisit the test plan.  There are  examples in aad.testplan.yaml for using RunCached.  MS.AAD.3.4v1's use of command, UpdateProviderExport, may be a good starting place.  Happy to talk if you would like a little more insight.  There are some limitations but many uses cases are covered by the RunCached approach.

Reply:
Thank you for the help! 2.2 should be fixed with 'RunCached'